File: Backend/TrackrAPI/TrackrAPI/Models/Acceso.cs
using System;
using System.Collections.Generic;

namespace TrackrAPI.Models
{
    public partial class Acceso
    {
        public Acceso()
        {
            AccesoPerfil = new HashSet<AccesoPerfil>();
            InverseIdAccesoPadreNavigation = new HashSet<Acceso>();
            JerarquiaAccesoEstructura = new HashSet<JerarquiaAccesoEstructura>();
        }

        public int IdAcceso { get; set; }
        public string Clave { get; set; } = null!;
        public string Nombre { get; set; } = null!;
        public string? Url { get; set; }
        public int? OrdenMenu { get; set; }
        public int IdTipoAcceso { get; set; }
        public int? IdAccesoPadre { get; set; }
        public int? IdIcono { get; set; }
        public string? Descripcion { get; set; }

        public virtual Acceso? IdAccesoPadreNavigation { get; set; }
        public virtual Icono? IdIconoNavigation { get; set; }
        public virtual TipoAcceso IdTipoAccesoNavigation { get; set; } = null!;
        public virtual ICollection<AccesoPerfil> AccesoPerfil { get; set; }
        public virtual ICollection<Acceso> InverseIdAccesoPadreNavigation { get; set; }
        public virtual ICollection<JerarquiaAccesoEstructura> JerarquiaAccesoEstructura { get; set; }
    }
}

File: Frontend/trackr-administrador/src/app/shared/models/seguridad/acceso.ts
export class Acceso {

  public idAcceso: number;
  public clave: string;
  public nombre: string;
  public ordenMenu?: number;
  public url: string;
  public idIcono?: number;
  public claseIcono: string;
  public idRolAcceso: number;
  public idTipoAcceso: number;
  public idAccesoPadre: number;
  public descripcion: string;

  public imagen: string;
  public urlImagen: string;
  public imagenBase64: string;
  public nombreImagen: string;
  public imagenTipoMime: string;
  public urlVideoAyuda: string;

  // Extras
  public hijos: Acceso[];
  public tipoAcceso: string;
  public claveRolAcceso: string;
  public claveTipoAcceso: string;
  public tieneAcceso: boolean;
  public cantidadDescendientes: number;

  constructor() {}
}

